todo_app/app.py

```diff
@@ -1,14 +1,69 @@
-from flask import Flask
-
+from flask import Flask, render_template, url_for, redirect
+from flask.globals import request
+from todo_app.data.session_items import get_item, get_items, add_item, save_item, remove_item
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
-
 @app.route('/')
-def index():
-    return 'Hello World!'
+def index(all=True):
+    return render_template('index.html', tasks = get_items(), all=all)
+
+@app.route('/sortby/<sort>')
```

Comment:
Having your sorting/filtering as a separate route has a couple of issues
- You could never filter and sort and the same time (e.g. sort by name & filter by status)
- You lose any selection whenever you submit any form

You could fix both of these issues if you instead
- always go to index, with a query parameter say if the user has just clicked on a filtering/sorting option
- use values on the `session` to store the user's sorting/filtering preference
- always redirect to index

Since the first issue isn't relevant yet, a simpler workaround for just the second issue is:
- Instead of redirecting to index at the end of each create/update/delete route, check for `request.referrer` and redirect to that if it's set.